From 3cbd2fc2d43bee266f85c44a739b06f1e3010493 Mon Sep 17 00:00:00 2001
From: rocka <i@rocka.me>
Date: Sat, 4 Nov 2023 23:29:35 +0800
Subject: [PATCH] Find Gettext and Fcitx5Module only when engine enabled. (#10)

* Find Gettext and Fcitx5Module only when engine enabled.

* Build translation only when ENABLE_ENGINE
---
 CMakeLists.txt | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b8e94e1..ee2bb79 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,8 +42,12 @@ set(LIBIME_JYUTPING_INSTALL_PKGDATADIR "${CMAKE_INSTALL_FULL_DATADIR}/libime")
 set(LIBIME_JYUTPING_INSTALL_LIBDATADIR "${CMAKE_INSTALL_FULL_LIBDIR}/libime")

 find_package(LibIMECore 1.0.16 REQUIRED)
-find_package(Gettext REQUIRED)
-find_package(Fcitx5Module REQUIRED Spell Punctuation QuickPhrase)
+
+if(ENABLE_ENGINE)
+    find_package(Gettext REQUIRED)
+    find_package(Fcitx5Module REQUIRED Spell Punctuation QuickPhrase)
+    add_subdirectory(po)
+endif()

 if(ENABLE_TEST)
     enable_testing()
@@ -53,7 +57,6 @@ endif()
 add_subdirectory(src)
 add_subdirectory(tools)
 add_subdirectory(data)
-add_subdirectory(po)

 if (ENABLE_DOC)
   find_package(Doxygen REQUIRED)
